File: tests/fixtures/csharp/members.cs
class Service
{
    private readonly Client client = Client.Create();
    private int first = LoadFirst(), second = LoadSecond();

    public event EventHandler? Changed, Completed;

    public event EventHandler Custom
    {
        add { Register(value); }
        remove { Unregister(value); }
    }

    public string Name { get; init; }

    public int this[int index]
    {
        get { return Load(index); }
        set { Save(index, value); }
    }

    public Service() : this(Build()) { }

    public Service(Client client) : base(client)
    {
        this.client = client;
    }

    ~Service() { Cleanup(); }

    public static Service operator +(Service left, Service right) => Merge(left, right);

    public static explicit operator int(Service value) => Convert(value);
}

File: tests/fixtures/php/members.php
<?php
class Configuration
{
    public string $name = "default", $mode;
    private static int $count = 0;
    public const FIRST = load_first(), SECOND = load_second();

    public function __construct(
        private Client $client,
        public readonly string $id = "default",
    ) {}

    public string $title {
        get => build_title();
        set => save_title($value);
    }
}
